ethereum/statetransition/src/main/java/tech/pegasys/teku/statetransition/forkchoice/BalanceAttackMitigationForkChoiceTrigger.java

```diff
@@ -42,11 +42,22 @@ public void onSlotStartedWhileSyncing(final UInt64 nodeSlot) {
 
   @Override
   public void onSlotStarted(final UInt64 nodeSlot) {
-    processHead(nodeSlot);
+    // We're technically processing attestations at the end of the previous slot so the fork choice
+    // slot needs to be nodeSlot - 1.  Otherwise we wind up deciding every slot is empty immediately
+    // and then treating the block as a reorg even if it arrives on time.
+    processHead(nodeSlot.minusMinZero(1));
```

Comment:
Should we subtract 1 from the slot in `onSlotStartedWhileSyncing` as well?

Reply:
Yes, done.